apps/system/js/utility_tray.js

```diff
@@ -46,22 +45,63 @@ window.UtilityTray = {
   focus: function() {},
   blur: function() {},
 
+  /**
+   * Update the max-height of the notifications container to ensure it doesn't
+   * overlap with the (position: fixed) footer.
+   */
+  recalculateNotificationsContainerHeight() {
+    this.notificationsContainer.style.maxHeight =
+      (this.footerContainer.offsetTop -
+       this.notificationsContainer.offsetTop) + 'px';
+  },
+
+  toggleFooterDisplay() {
+    var shouldShow = (this.motion.percentVisible === 1);
+    this.footer.classList.toggle('animate-in', shouldShow);
+    this.footer.classList.toggle('animate-out', !shouldShow);
+  },
+
   init: function ut_init() {
-    var touchEvents = [
-      'touchstart',
-      'touchmove',
-      'touchend',
-      'mousedown'
-    ];
-
-    touchEvents.forEach(function bindEvents(name) {
-      this.overlay.addEventListener(name, this);
-      this.statusbarIcons.addEventListener(name, this);
-      this.grippy.addEventListener(name, this);
-      this.topPanel.addEventListener(name, this);
-    }, this);
+
+    // Prepare the tray's motion behavior.
+    this.motion = new UtilityTrayMotion(this.motionElement);
+
+    this.motion.el.addEventListener('tray-motion-state', this);
+    this.motion.el.addEventListener('tray-motion-moved',
```

Comment:
Did some testing on a flame, it's still a lot better than master but you can feel a difference when removing all the scroll/touch listening and adding basic (but a bit buggy) scroll-snapping.
Not arguing we should do that, but I'd like to see if we can unload the event loop a little.

One way to do that would be to forgo the `tray-motion-moved` event. I think we should be able to toggle the footer display based on `tray-motion-state` events only.

What do you think?